tool/make-native-distribution.sh

```diff
@@ -52,9 +58,20 @@ PREFIX=$(cd "$PREFIX" && pwd -P)
 mkdir -p "$PREFIX/build"
 cd "$PREFIX/build"
 
-git clone --branch $TAG "$(mx urlrewrite https://github.com/oracle/truffleruby.git)"
-git clone --branch $TAG "$(mx urlrewrite https://github.com/graalvm/sulong.git)"
-git clone --branch $TAG "$(mx urlrewrite https://github.com/oracle/graal.git)"
+if [ -n "$TAG" ]; then
+  git clone --branch $TAG "$(mx urlrewrite https://github.com/oracle/truffleruby.git)"
+  git clone --branch $TAG "$(mx urlrewrite https://github.com/graalvm/sulong.git)"
+  git clone --branch $TAG "$(mx urlrewrite https://github.com/oracle/graal.git)"
+else
+  git clone "$original_repo" truffleruby
+
+  if [ -d "$original_repo/../graal" ] && [ -d "$original_repo/../sulong" ]; then
+    # Building locally (not in CI), copy from local repositories to gain time
+    git clone "$original_repo/../graal" graal
+    git clone "$original_repo/../sulong" sulong
```

Comment:
Shallow clones (`--depth 1`) could speed things up if they are throw away clones.

Reply:
If we don't do it here, `mx` will clone (on the first mx command, here `mx sversions`).
Also, we need to clone specific revisions which have no branch so shallow clones won't work.

Reply:
But you could shallow clone if a tag is provided.

Reply:
True, for the logic in the other branch.

Reply:
5ddd89bab8dbbe714c36ca46e653dfebb44f97c4